google-cloud-logging/lib/google/cloud/logging/middleware.rb

```diff
@@ -57,18 +57,19 @@ def initialize app, logger: nil, **kwargs
 
           load_config kwargs
 
-          if logger
-            @logger = logger
-          else
+          logger ||= Middleware.logger
+          logger ||= begin
             log_name = configuration.log_name
             logging = Logging.new project_id: configuration.project_id,
                                   credentials: configuration.credentials
             resource = Middleware.build_monitored_resource(
               configuration.monitored_resource.type,
               configuration.monitored_resource.labels
             )
-            @logger = logging.logger log_name, resource
+            Middleware.logger = logging.logger log_name, resource
           end
+
```

Comment:
The middleware isn't currently setting the Rails logger when it is initialized or when it is called. Should it? Currently, if the user doesn't set `set_default_logger_on_rails_init` to `true`, or doesn't call `Google::Cloud::Logging::Railtie.set_default_logger` in a post-fork hook, it never does get set. Is it strange to have the Rack middleware call a Rails integration helper method?

Reply:
Yes we need to have the middleware set the Rails logger if the user didn't do so explicitly through some other means. And yes, it is possible for the middleware to be used without Rails, so it should not call the Rails method directly. Might need the Rails initializer to pass some kind of callback into the middleware constructor to avoid the coupling.

Reply:
A callback is a great idea. Working on it.

Reply:
This has been added. PTAL.